src/pykoala/cubing.py

```diff
@@ -570,29 +570,32 @@ class Cube(SpectraContainer):
     n_rows = None
     x_size_arcsec = None
     y_size_arcsec = None
-
-    def __init__(self, hdul=None, file_path=None, 
-                 hdul_extensions_map=None, **kwargs):
+    hdul_extensions_map = {"INTENSITY": "INTENSITY", "VARIANCE": "VARIANCE"}
+    
+    def __init__(self, hdul=None, hdul_extensions_map=None, **kwargs):
 
         self.hdul = hdul
-        self.hdul_extensions_map = hdul_extensions_map
 
-        if self.hdul_extensions_map is None:
-            self.hdul_extensions_map = {"INTENSITY": "INTENSITY",
-                                        "VARIANCE": "VARIANCE"}
-        if self.hdul is not None:
-            print("[Cube] Initialising cube from input HDUL")
-            self.hdul = hdul
-        elif file_path is not None:
-            self.load_hdul(file_path)
-        self.get_wcs_from_header()
+        if hdul_extensions_map is not None:
+            self.hdul_extensions_map = hdul_extensions_map
+
         super().__init__(intensity=self.intensity,
                          variance=self.variance,
                          **kwargs)
+        self.get_wcs_from_header()
         self.parse_info_from_header()
         self.n_wavelength, self.n_rows, self.n_cols = self.intensity.shape
         self.get_wavelength()
 
+    @property
+    def hdul(self):
+        return self._hdul
+
+    @hdul.setter
+    def hdul(self, hdul):
+        assert isinstance(hdul, fits.HDUList)
+        self._hdul = hdul
+
     @property
     def intensity(self):
         return self.hdul[self.hdul_extensions_map['INTENSITY']].data
```

Comment:
This seems a good idea, somewhat different from the behaviour of `RSS`. Should we implement the `hdul_extensions_map` scheme there?

Alternatively, `intensity` could always be an array, and the `hdul_extensions_map` would only be relevant when [reading and] writing FITS files.

Reply:
Indeed, we need to create an standard format for the RSS files processed with pykoala. Including the map sounds reasonable.

Let's see if this argument is convincing. Making `intensity` an array implies to allocate the memory space by default. Instead, if the data is read from the HDUL, only if when calling variable the memory is allocated...In any case, this is still a part of the architecture that is poorly defined, and I believe we will end up modifying it.